eclipse-scout-core/src/dataobject/serialize/IdDoNodeSerializer.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2010, 2024 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+import {Constructor, DoDeserializer, DoNodeSerializer, DoSerializer, DoValueMetaData, Id, IdParser, scout} from '../../index';
+
+export class IdDoNodeSerializer implements DoNodeSerializer<Id<any>> {
+
+  protected static _ID_PARSER: IdParser; // created on first use. Access using IdDoNodeSerializer._getIdParser()
+
+  canSerialize(value: any, metaData: DoValueMetaData): boolean {
+    return value instanceof Id;
+  }
+
+  serialize(value: Id<any>, metaData: DoValueMetaData, serializer: DoSerializer): any {
+    // It's not possible to perfectly compute if the backend expects a qualified or unqualified id (if the Java DoNode is concrete or not, see ScoutDataObjectSerializerProvider.java).
+    // Therefore, always serialize qualified, which might include too much information.
+    // If the Java deserializer uses unqualified parsing, the unnecessary extra information is removed (see IdCodec.java#fromUnqualifiedUnchecked).
+    return value.toQualified();
+  }
+
+  canDeserialize(value: any, metaData: DoValueMetaData): boolean {
+    if (!(typeof value === 'string')) {
+      return false;
+    }
+    if (!metaData?.type) {
+      return false;
+    }
+    return this.isIdClass(metaData.type) || Id.isPrototypeOf(metaData.type);
+  }
+
+  deserialize(rawId: string, metaData: DoValueMetaData, deserializer: DoDeserializer): Id<any> {
+    return IdDoNodeSerializer._getIdParser().parse(metaData.type, rawId, () => this.detectConcreteTypeName(metaData));
+  }
+
+  isIdClass(candidate: Constructor) {
```

Comment:
Missing return type.

Also: what is the expected ts-error here?

Reply:
Correct. Fixed.

The ts error is TS2367: unintentional comparison.